vault/data_source_ldap_static_role_credentials.go

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package vault
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"log"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+	"github.com/hashicorp/terraform-provider-vault/internal/provider"
+	"github.com/hashicorp/vault/api"
+)
+
+func ldapStaticCredDataSource() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: ReadContextWrapper(readLDAPStaticCreds),
+		Schema: map[string]*schema.Schema{
+			consts.FieldPath: {
```

Comment:
TFVP has inconsistent naming for this field. But I think we should start to enforce consistency from here on out. In the spirit of that, I would recommend changing this to FieldMount like @vinay-gopalan does [here](https://github.com/hashicorp/terraform-provider-vault/pull/1829/files#diff-4c9591a10e0db1730505bb6b3e3c11e4cb9eccc773e71cea05a320999e79ec6cR37).

So for all non-backend-mounting resources we can use FieldMount. For all backend-mounting resources, like our  `vault_ldap_secret_backend` we can use FieldPath. What do you think?

Reply:
+1 on setting a precedent with this pattern. This also ties in with Vault's terminology, since what this field represents is where the mount lives, and these resources live in that mount. 

The idea is that we can use `FieldPath` to describe the path where an engine will be mounted. We can deprecate the use of `FieldBackend` for any of these scenarios and document that it is no longer part of the desired code style. Open to any thoughts and suggestions! 😄